### PR TITLE
Added button special information to game screens

### LINKED ISSUE #1200: Button specials need descriptions
Some button specials have been implemented already. Their ```BMBtnSkill``` classes should have a static ```get_description()``` method on them that returns the description of that special as a string. (And as new ones are implemented, they should be given descriptions as well.)

### PR DESCRIPTION
**Pull request by blackshadowshade**
Fixes #1200.

I have added the button skills before the die skills, but not otherwise discriminated them, because this was the easiest thing to implement.

Tested with Giant vs RandomBMMixed, Echo vs Temperance, and Washu vs Shadow Warriors.

http://jenkins.buttonweavers.com:8080/job/buttonmen-blackshadowshade/834/

### CODE REVIEW

**Comment by cgolubi1** [comment]
This looks quite reasonable, and i don't think there's any need to play RandomAI games and look for internal errors or anything, so i'll just bring it up and see how it looks, covering each of the cases of:
* Game has no button skills or die skills
* Game has button skills and no die skills
* Game has no button skills, and die skills
* Game has button skills and die skills
* Game has more than one different button skill
* Game has more than one identical button skill

I'll look for any weirdness in the apache error log, and call that good enough.

**Comment by cgolubi1** [comment]
Hmm, why does the mouseover text for Giant say "Giant (Giant): Cannot win initiative"?  That doesn't seem right.

**Comment by cgolubi1** [comment]
Nothing in the error log.  So the only issue i see is the "Giant (Giant)" thing.

File: src/engine/BMBtnSkill.php [hunk 1]
```diff
@@ -20,4 +20,34 @@ class BMBtnSkill {
...
+            }
+
+            $skillInstance = new $skillClass;
+            if ($skillInstance instanceof BMBtnSkillArtificialReenable) {
+                return NULL;
+            }
+        }
+
+        $skillDescription = array(
+            'code' => $skill,
```
Comment (cgolubi1) [inline src/engine/BMBtnSkill.php]: Yeah, okay, now i understand what you're doing here that's leading to the doubled "Giant", and it's definitely a bug.  I guess it happens that the UI doesn't assume codes are single-character strings found in recipes, but there's no reason someone couldn't make that assumption in using API output, so i don't think it makes sense to break that.

I think your easiest solution is to leave 'code' empty (or not pass it), and have the UI not display the parens if there is no code.  If you wanted to pass a new parameter (e.g. 'type' => 'die' or 'type' => 'button'), you could, but that would be a lot more test updating, so i don't think you need to.  I think overloading 'code' is a bug.

**Comment by blackshadowshade** [comment]
Okay, done. Tested with Giant vs Echo.

http://jenkins.buttonweavers.com:8080/job/buttonmen-blackshadowshade/841/ (if it passes)

**Comment by cgolubi1** [comment]
Looks right to me based on the code.  Checking it out using the test games i made for the previous testing.

**Pull request merged**